### DISCUSSION TITLE
Deleting a collection causing a graphQL error

### DISCUSSION THREAD

**Discussion by iFrogHop2Worlds** [tinacms/tinacms#4143] [Help]
Hey so just want start off with I could be totally going the wrong way with how I am approaching this but I am setting up the cms right now so that each page is running off it's own collection and maybe inheriting from other collections. The Page collections are meant to be like theme/layouts which can be saved and reused. But when I delete a page collection I get this grapgQl error
```
GraphQLError [Object]: Unable to find record content/home/New_Layout.md
    at Database.get (C:\source\mtfurniture\node_modules\@tinacms\graphql\dist\index.js:5110:17)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Resolver.getDocumentOrDirectory (C:\source\mtfurniture\node_modules\@tinacms\graphql\dist\index.js:4323:23)
    at async C:\source\mtfurniture\node_modules\@tinacms\graphql\dist\index.js:5587:26
    at async reducePromises (C:\source\mtfurniture\node_modules\@tinacms\graphql\dist\index.js:72:18)
    at async sequential (C:\source\mtfurniture\node_modules\@tinacms\graphql\dist\index.js:78:18)
    at async Database.query (C:\source\mtfurniture\node_modules\@tinacms\graphql\dist\index.js:5585:16)
    at async Resolver.resolveCollectionConnection (C:\source\mtfurniture\node_modules\@tinacms\graphql\dist\index.js:4688:22)
    at async Promise.all (index 4)
    at async Promise.all (index 0)
    at async resolve (C:\source\mtfurniture\node_modules\@tinacms\graphql\dist\index.js:4853:17)
    at async C:\source\mtfurniture\node_modules\@tinacms\cli\dist\index.js:1211:26
Error querying file content/home/New_Layout.md from collection Landing_Page. Please run "tinacms audit" or add the --verbose option  for more info

GraphQL request:8:11
8 |           documents(sort: $sort, after: $after, first: $limit, filter: $filter,
  |           ^
  | folder: $folder) @include(if: $includeDocuments) {
```

Am I missing something? Should I be making the page layouts as templates inside a single collection? Or is this doable by creating a new collection per layout/content for each page?

**Comment by jamespohalloran** [comment]
You can model your setup to have one collection per page if you want, but just note that editors won't be able to create/delete collections. They can create/delete individual documents in a collection.

> GraphQLError [Object]: Unable to find record content/home/New_Layout.md

The error seems like somewhere in your site, you're requesting `home/New_Layout.md` (from a specific collection), and that collection/document no longer exists.

**Reply by iFrogHop2Worlds** [reply]
The thing is I am only pulling in 1 collection per page. Im not referencing the other collections in anyway yet in my code. When I delete a collection from the cms panel the graphQL api still expecting to find the .MD file. It's not my code calling causing the error. Ill test this in production maybe its just a local dev issue?

**Reply by jamespohalloran** [reply]
I'm not sure I understand. Seeing a reproducable repo, or a loom video might help. 

If you're getting this error:

> GraphQLError [Object]: Unable to find record content/home/New_Layout.md

Something in your site must be requesting that document, right?

**Reply by jerefrer** [reply]
Same problem reported as an actual issue here : https://github.com/tinacms/tinacms/issues/4150
I don't think @iFrogHop2Worlds is requesting anything anywhere in his code, this happens within Tina Admin itself.

**Reply by iFrogHop2Worlds** [reply]
My solution for now was to just set creation and deletion to false on my schema. Which is not a big issue in this case but in the future it would be if this persists to production which I haven't actually tried yet. Sort in vacation mode atm

@jerefrer did you get a chance to try this in production? Is this only a local development issue?

@jamespohalloran Seems like this is a windows issue with graphql on tina admin. Myself and 2 others experiencing this all using windows.
